stacks_queues/queue.py

```diff
@@ -15,48 +16,74 @@ def __init__(self):
         self.front = -1
         self.rear = -1
         self.size = 0
-      
+
 
     def enqueue(self, element):
         """ Adds an element to the Queue
             Raises a QueueFullException if all elements
             In the store are occupied
             returns None
         """
-        pass
+        if self.size == self.buffer_size:
+            raise QueueFullException()
+        if self.size == 0:
+            self.front = 0
+        self.rear = (self.rear + 1) % self.buffer_size
+        self.store[self.rear] = element
+        self.size = self.size + 1
+    
+        
 
     def dequeue(self):
         """ Removes an element from the Queue
             Raises a QueueEmptyException if 
             The Queue is empty.
             returns None
         """
-        pass
+        if self.size == 0:
+            raise QueueEmptyException()
+
+    
+        tmp = self.store[self.front]
+        self.front = (self.front + 1) % self.buffer_size
+        self.size = self.size - 1
+        return tmp
 
     def front(self):
         """ Returns an element from the front
             of the Queue and None if the Queue
             is empty.  Does not remove anything.
         """
-        pass
+        if self.empty():
+            return None
+        return self.store[self.front]
         
 
     def size(self):
         """ Returns the number of elements in
             The Queue
         """
-        pass
+        return self.size
 
     def empty(self):
         """ Returns True if the Queue is empty
             And False otherwise.
         """
-        pass
+        if self.size == 0:
+            return True
+        
+        return False
```

Comment:
This could be a bit simpler

```suggestion
        return self.size == 0
```